datalad/interface/base.py

```diff
@@ -329,24 +330,25 @@ def _prep(
 
         Paths are then sorted by the datasets that contain them. If paths are
         detected that are not associated with any dataset `ValueError` is
-        raised.
+        raised. If a `dataset` is given, any paths associated with a dataset
+        that is not this dataset or a subdataset of it will also trigger a
+        `ValueError`.
```

Comment:
Well, if that is the case, we might just want to make that attribute of `PathOutsideRepositoryError` optional. However, I'm satisfied with an issue referencing this piece of code ;-)